homeassistant/components/forked_daapd/media_player.py

```diff
@@ -0,0 +1,673 @@
+                    i
+                    for i, item in enumerate(saved_queue["items"])
+                    if item["id"] == self._player["item_id"]
+                )
+                await self._api.clear_queue()
+            self._tts_requested = True
+            await self._api.add_to_queue(uris=media_id, playback="start")
+            try:
+                await asyncio.wait_for(
+                    self._tts_playing_event.wait(), timeout=TTS_TIMEOUT
+                )
+                # we have started TTS, now wait for completion
+                await asyncio.sleep(
+                    self._queue["items"][0]["length_ms"]
+                    / 1000  # player may not have updated yet so grab length from queue
+                    + self._tts_pause_time
+                )
+            except asyncio.TimeoutError:
+                self._tts_requested = False
+                _LOGGER.warning("TTS request timed out.")
+            self._tts_playing_event.clear()
+            # TTS done, return to normal
+            await self.async_turn_on()  # restores outputs
+            await self._api.clear_queue()
+            if self._pipe_control_api:  # resume pipe
+                if saved_state == STATE_PLAYING:
+                    await self.async_media_play()
+            else:  # restore stashed queue
+                if saved_queue:
+                    uris = ""
+                    for item in saved_queue["items"]:
+                        uris += item["uri"] + ","
+                    await self._api.add_to_queue(
+                        uris=uris,
+                        playback="start",
+                        playback_from_position=saved_queue_position,
+                    )
+                    await self._api.seek(position_ms=saved_song_position)
+                    if saved_state != STATE_PLAYING:
+                        await self.async_media_pause()
+        else:
+            _LOGGER.debug("Media type '%s' not supported.", media_type)
+
+
+class ForkedDaapdUpdater:
+    """Manage updates for the forked-daapd device."""
+
+    def __init__(self, hass, api):
+        """Initialize."""
+        self.hass = hass
+        self._api = api
+        self.websocket_handler = None
+        self._all_output_ids = []
+
+    async def async_init(self):
+        """Perform async portion of class initialization."""
+        server_config = await self._api.get_request("config")
+        websocket_port = server_config.get("websocket_port")
+        if websocket_port:
+            self.websocket_handler = asyncio.create_task(
+                self._api.start_websocket_handler(
+                    server_config["websocket_port"],
+                    WS_NOTIFY_EVENT_TYPES,
+                    self._update,
+                    WEBSOCKET_RECONNECT_TIME,
+                    self._disconnected_callback,
+                )
+            )
+        else:
+            _LOGGER.error("Invalid websocket port.")
```

Comment:
We don't end logging messages with period.
```suggestion
            _LOGGER.error("Invalid websocket port")
```